js/plates-model/field.js

```diff
@@ -263,14 +291,14 @@ export default class Field extends FieldBase {
         // Special case when the continent is "trying" to subduct under the ocean. Apply drag force to stop both plates.
         this.draggingPlate = field.plate
       }
-    } else if (this.isContinent && field.isContinent && !field.island) {
+    } else if (this.isContinent && field.isContinent) {
```

Comment:
It seems like the effect of this refactor is that old checks for `isContinent` should become checks for `continentalCrust` and `isContinent && !isIsland` should become `isContinent`. Following that logic, I would expect this check to read `else if (this.continentalCrust && field.isContinent)`

Reply:
Good catch! Although, I think that `&& !field.island` was unnecessary. Island-continent collision is handled above and this specific collision type is only applied to "real" continents (orogeny / mountains forming).

btw, another candidate for refactoring is this collision logic. I guess it would be nice to get rid of `else` key word and make sure that all the conditions are self-sufficient and don't depend on the others. But then it'd be nice to check if it affects performance in a visible way or not.

Reply:
Done in this commit: 
https://github.com/concord-consortium/plate-tectonics-3d/commit/54ac5f9a13aabbe23f6914ae8d51db0e652c88e5

I hope it looks cleaner now and also it's faster as we limit the number of checks. It affects model behavior a bit, but it doesn't seem that it's better or worse. It's just a bit different.